src/constraint.rs

```diff
@@ -734,14 +734,14 @@ impl ContainsConstraint {
 impl ConstraintValidator for ContainsConstraint {
     fn validate(&self, value: &OwnedElement, type_store: &TypeStore) -> ValidationResult {
         // get the expected values from the contains constraint
-        let mut expected_values = self.values.iter();
+        let mut expected_values = self.values.to_owned();
```

Comment:
Does this give you a copy of the data, or ownership of the original? If the original data gets mutated, it's going to lead to weird bugs.